src/OAuth/OAuth2/Service/Bitrix24.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace OAuth\OAuth2\Service;
+
+use OAuth\OAuth2\Token\StdOAuth2Token;
+use OAuth\Common\Http\Exception\TokenResponseException;
+use OAuth\Common\Http\Uri\Uri;
+use OAuth\Common\Consumer\CredentialsInterface;
+use OAuth\Common\Http\Client\ClientInterface;
+use OAuth\Common\Storage\TokenStorageInterface;
+use OAuth\Common\Http\Uri\UriInterface;
+
+class Bitrix24 extends AbstractService
+{
+    const SCOPE_DEPARTMENT = 'department';
+    const SCOPE_CRM = 'crm';
+    const SCOPE_CALENDAR = 'calendar';
+    const SCOPE_USER = 'user';
+    const SCOPE_ENTITY = 'entity';
+    const SCOPE_TASK = 'task';
+    const SCOPE_TASKS_EXTENDED = 'tasks_extended';
+    const SCOPE_IM = 'im';
+    const SCOPE_LOG = 'log';
+    const SCOPE_SONET_GROUP = 'sonet_group';
+
+    public function __construct(
+        CredentialsInterface $credentials,
+        ClientInterface $httpClient,
+        TokenStorageInterface $storage,
+        array $scopes,
+        UriInterface $baseApiUri
+    ) {
+        parent::__construct($credentials, $httpClient, $storage, $scopes, $baseApiUri);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAuthorizationEndpoint()
+    {
+        return new Uri(sprintf('%s/oauth/authorize/', $this->baseApiUri));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAccessTokenEndpoint()
+    {
+        return new Uri(sprintf('%s/oauth/token/', $this->baseApiUri));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requestAccessToken($code, $state = null)
+    {
+        if (null !== $state) {
+            $this->validateAuthorizationState($state);
+        }
+
+        $responseBody = $this->httpClient->retrieveResponse(
+            $this->getAccessTokenUri($code),
+            array(),
+            $this->getExtraOAuthHeaders(),
+            'GET'
+        );
+
+        $token = $this->parseAccessTokenResponse($responseBody);
+        $this->storage->storeAccessToken($this->service(), $token);
+
+        return $token;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAccessTokenUri($code)
+    {
+        $parameters = array(
+            'code'          => $code,
+            'client_id'     => $this->credentials->getConsumerId(),
+            'client_secret' => $this->credentials->getConsumerSecret(),
+            'redirect_uri'  => $this->credentials->getCallbackUrl(),
+            'grant_type'    => 'authorization_code',
+            'scope'         => $this->scopes
+        );
+
+        $parameters['scope'] = implode(' ', $this->scopes);
+
+        // Build the url
+        $url = clone $this->getAccessTokenEndpoint();
```

Comment:
You don't need to clone the object as `getAccessTokenEndpoint()` returns a new new object.